### DISCUSSION TITLE
Conditional Draft Creation

### DISCUSSION THREAD

**Discussion by dannoh** [OrchardCMS/OrchardCore#8008] [Q&A]
Piggy backing on this closed issue a bit  #6749 
If we have a ContentItem with 3 properties.  Name, Description, Content.  We would like to create a new draft if Name or Content are changed.  However, if the user changes the Description we would like to just update the Published version without creating a draft.

Is this at all possible with the existing code?

**Comment by dannoh** [comment]
Originally posted this to General accidentally, hoping a move to Q&A might get an answer.

**Comment by deanmarcussen** [comment]
No this is not possible with the current code. You would have to implement a custom content manager I imagine

**Comment by hishamco** [comment]
What if we hook the content updating event

**Comment by Skrypt** [comment]
Right now the DefaultContentManager defines what happens when someone saves a draft or publish a content item. Those "buttons" will eventually use the DefaultContentManager methods to save content items in the database AND the OrchardCore.Contents module main controller is tightly integrated with this DefaultContentManager. So, I believe that if you want to do what you want ; you could customize it. Easy way would be to simply copy the OrchardCore.Contents module and rename it. Change the main controller and DefaultContentManager to suit your own needs. Though, all this at your own risks.

Some ideas : 

If we'd want to make this more Generic in Orchard Core right now there is nothing that does this. So, basically what we would want is a conditional event execution based on rules. What we would need to develop then is a module that would add conditional rules on all Events triggered by the "DefaultContentManager" so that they could break the current code execution. It's worth investigating if this could be possible. But for achieving this, it's also conditional on what we do in the DefaultContentManager methods. Because we often call the events after everything is "saved" in the DB, though, the db _session operations are no-op in the DefaultContentManager, so I definitely think this could be achieved through extending events with handlers but that might not be perfect or easy to do.

Also, looking back at your request. Then these rules would also affect which event should be triggered which makes this really more complex than just using conditional rules on events. 

**This should be added directly in the Controller actions to evaluate the rules and then execute the proper DefaultContentManager methods based on that.**
So scratch everything I just wrote before basically 😄

**Comment by Skrypt** [comment]
Or maybe simply use Javascript to disable those buttons based on the conditions you require.
From a custom controller you call BuildEditorAsync "to create your viewModel" and you add your own custom javascript that will add validations on these specific fields.

Lots of different methods to get there for sure.
